diff --git a/gfx/wr/webrender/src/renderer/init.rs b/gfx/wr/webrender/src/renderer/init.rs
index d64a8a2a9f20..18da8cfd66df 100644
--- a/gfx/wr/webrender/src/renderer/init.rs
+++ b/gfx/wr/webrender/src/renderer/init.rs
@@ -4,7 +4,7 @@
 
 use api::{BlobImageHandler, ColorF, CrashAnnotator, DocumentId, IdNamespace};
 use api::{RenderBackendId, VoidPtrToSizeFn, FontRenderMode, ImageFormat};
-use api::{RenderNotifier, ImageBufferKind};
+use api::{RenderNotifier, ImageBufferKind, NotificationRequest};
 use api::units::*;
 use api::channel::unbounded_channel;
 pub use api::DebugFlags;
@@ -112,9 +112,13 @@ pub trait AsyncPropertySampler {
     /// and before it processes anything.
     fn register(&self);
     /// This is called for each transaction with the generate_frame flag set
-    /// (i.e. that will trigger a render). The list of frame messages returned
-    /// are processed as though they were part of the original transaction.
-    fn sample(&self, document_id: DocumentId, generated_frame_id: Option<u64>) -> Vec<FrameMsg>;
+    /// (i.e. that will trigger a render). The frame messages and notifications
+    /// returned are processed as though they were part of the original transaction.
+    fn sample(
+        &self,
+        document_id: DocumentId,
+        generated_frame_id: Option<u64>,
+    ) -> (Vec<FrameMsg>, Vec<NotificationRequest>);
     /// This is called exactly once, when the render backend thread is about to
     /// terminate.
     fn deregister(&self);
