diff --git a/dom/ipc/BrowserParent.h b/dom/ipc/BrowserParent.h
index 876f0c6bc0b5..83ce854eb4be 100644
--- a/dom/ipc/BrowserParent.h
+++ b/dom/ipc/BrowserParent.h
@@ -11,6 +11,7 @@
 #include "mozilla/AlreadyAddRefed.h"
 #include "mozilla/ContentCache.h"
 #include "mozilla/EventForwards.h"
+#include "mozilla/NativeTextInput.h"
 #include "mozilla/RefPtr.h"
 #include "mozilla/dom/BrowserBridgeParent.h"
 #include "mozilla/dom/PBrowserParent.h"
@@ -397,6 +398,7 @@ class BrowserParent final : public PBrowserParent,
       const nsCursor& aValue, Maybe<IPCImage>&& aCustomCursor,
       const float& aResolutionX, const float& aResolutionY,
       const uint32_t& aHotspotX, const uint32_t& aHotspotY, const bool& aForce);
+  mozilla::ipc::IPCResult RecvUpdateBackgroundColor(const nscolor& aColor);
 
   mozilla::ipc::IPCResult RecvSetLinkStatus(const nsString& aStatus);
 
@@ -612,6 +614,10 @@ class BrowserParent final : public PBrowserParent,
 
   bool HandleQueryContentEvent(mozilla::WidgetQueryContentEvent& aEvent);
 
+  bool QueryNativeTextInput(const mozilla::NativeTextInputRequest& aRequest,
+                            mozilla::NativeTextInputReply& aReply);
+  mozilla::ipc::IPCResult RecvNativeTextInputPositionChanged();
+
   bool SendSimpleContentCommandEvent(
       const mozilla::WidgetContentCommandEvent& aEvent);
   bool SendInsertText(const mozilla::WidgetContentCommandEvent& aEvent);
@@ -674,7 +680,7 @@ class BrowserParent final : public PBrowserParent,
 
   void LayerTreeUpdate(bool aActive);
 
-  mozilla::ipc::IPCResult RecvInvokeDragSession(
+  MOZ_CAN_RUN_SCRIPT_BOUNDARY mozilla::ipc::IPCResult RecvInvokeDragSession(
       nsTArray<IPCTransferableData>&& aTransferables, const uint32_t& aAction,
       Maybe<BigBuffer>&& aVisualDnDData, const uint32_t& aStride,
       const gfx::SurfaceFormat& aFormat, const LayoutDeviceIntRect& aDragRect,
@@ -970,6 +976,10 @@ class BrowserParent final : public PBrowserParent,
   // tab, it should take this appearance.
   nsIWidget::Cursor mCursor;
 
+#ifdef XP_IOS
+  Maybe<nscolor> mBackgroundColor;
+#endif
+
   nsTArray<nsString> mVerifyDropLinks;
 
 #ifdef DEBUG
