diff --git a/gfx/layers/ipc/UiCompositorControllerParent.cpp b/gfx/layers/ipc/UiCompositorControllerParent.cpp
index b04c706d6845..9340224d6870 100644
--- a/gfx/layers/ipc/UiCompositorControllerParent.cpp
+++ b/gfx/layers/ipc/UiCompositorControllerParent.cpp
@@ -112,16 +112,17 @@ mozilla::ipc::IPCResult UiCompositorControllerParent::RecvMaxToolbarHeight(
   return IPC_OK();
 }
 
-mozilla::ipc::IPCResult UiCompositorControllerParent::RecvFixedBottomOffset(
-    const int32_t& aOffset) {
-#if defined(MOZ_WIDGET_ANDROID)
+mozilla::ipc::IPCResult UiCompositorControllerParent::RecvFixedLayerMargins(
+    const int32_t& aTop, const int32_t& aBottom, const int32_t& aTopInset,
+    const int32_t& aBottomInset) {
+#if defined(MOZ_WIDGET_ANDROID) || defined(MOZ_WIDGET_UIKIT)
   CompositorBridgeParent* parent =
       CompositorBridgeParent::GetCompositorBridgeParentFromLayersId(
           mRootLayerTreeId);
   if (parent) {
-    parent->SetFixedLayerMargins(0, aOffset);
+    parent->SetFixedLayerMargins(aTop, aBottom, aTopInset, aBottomInset);
   }
-#endif  // defined(MOZ_WIDGET_ANDROID)
+#endif
 
   return IPC_OK();
 }
@@ -232,6 +233,14 @@ void UiCompositorControllerParent::NotifyCompositorScrollUpdates(
           std::move(aUpdates)));
 }
 
+void UiCompositorControllerParent::NotifyAsyncScrollStateChanged(
+    nsTArray<ScrollableLayerGuid>&& aScrollingLayers) {
+  CompositorThread()->Dispatch(NewRunnableMethod<nsTArray<ScrollableLayerGuid>>(
+      "UiCompositorControllerParent::SendAsyncScrollStateChanged", this,
+      &UiCompositorControllerParent::SendAsyncScrollStateChanged,
+      std::move(aScrollingLayers)));
+}
+
 UiCompositorControllerParent::UiCompositorControllerParent(
     const LayersId& aRootLayerTreeId)
     : mRootLayerTreeId(aRootLayerTreeId)
