diff --git a/gfx/layers/ipc/UiCompositorControllerParent.h b/gfx/layers/ipc/UiCompositorControllerParent.h
index 5f737ba747ac..dcc7c373d197 100644
--- a/gfx/layers/ipc/UiCompositorControllerParent.h
+++ b/gfx/layers/ipc/UiCompositorControllerParent.h
@@ -39,13 +39,18 @@ class UiCompositorControllerParent final
                                               bool* aOutResumed);
   mozilla::ipc::IPCResult RecvInvalidateAndRender();
   mozilla::ipc::IPCResult RecvMaxToolbarHeight(const int32_t& aHeight);
-  mozilla::ipc::IPCResult RecvFixedBottomOffset(const int32_t& aOffset);
+  mozilla::ipc::IPCResult RecvFixedLayerMargins(const int32_t& aTop,
+                                                const int32_t& aBottom,
+                                                const int32_t& aTopInset,
+                                                const int32_t& aBottomInset);
   mozilla::ipc::IPCResult RecvDefaultClearColor(const uint32_t& aColor);
   mozilla::ipc::IPCResult RecvRequestScreenPixels(
       uint64_t aRequestId, gfx::IntRect aSourceRect,
       ipc::FileDescriptor&& aHardwareBuffer);
   mozilla::ipc::IPCResult RecvEnableLayerUpdateNotifications(
       const bool& aEnable);
+  mozilla::ipc::IPCResult RecvNativeTextInputScrollers(
+      const nsTArray<ScrollableLayerGuid>& aScrollers);
   void ActorDestroy(ActorDestroyReason aWhy) override;
 
   // Class specific functions
