diff --git a/dom/events/ContentEventHandler.h b/dom/events/ContentEventHandler.h
index e383b6c7c97d..4ad82617e4c3 100644
--- a/dom/events/ContentEventHandler.h
+++ b/dom/events/ContentEventHandler.h
@@ -22,6 +22,9 @@ struct nsRect;
 
 namespace mozilla {
 
+struct NativeTextInputRequest;
+struct NativeTextInputReply;
+
 namespace dom {
 class Element;
 }  // namespace dom
@@ -128,6 +131,9 @@ class MOZ_STACK_CLASS ContentEventHandler {
 
   explicit ContentEventHandler(nsPresContext* aPresContext);
 
+  MOZ_CAN_RUN_SCRIPT nsresult QueryNativeTextInput(
+      const NativeTextInputRequest& aRequest, NativeTextInputReply& aReply);
+
   // Handle aEvent in the current process.
   MOZ_CAN_RUN_SCRIPT nsresult
   HandleQueryContentEvent(WidgetQueryContentEvent* aEvent);
@@ -147,7 +153,8 @@ class MOZ_STACK_CLASS ContentEventHandler {
   OnQueryTextRectArray(WidgetQueryContentEvent* aEvent);
   // eQueryEditorRect event handler
   MOZ_CAN_RUN_SCRIPT nsresult
-  OnQueryEditorRect(WidgetQueryContentEvent* aEvent);
+  OnQueryEditorRect(WidgetQueryContentEvent* aEvent,
+                    NativeTextInputReply* aNativeGeometry = nullptr);
   // eQueryContentState event handler
   MOZ_CAN_RUN_SCRIPT nsresult
   OnQueryContentState(WidgetQueryContentEvent* aEvent);
@@ -165,7 +172,9 @@ class MOZ_STACK_CLASS ContentEventHandler {
   OnQueryDropTargetHittest(WidgetQueryContentEvent* aEvent);
 
   // NS_SELECTION_* event
-  MOZ_CAN_RUN_SCRIPT nsresult OnSelectionEvent(WidgetSelectionEvent* aEvent);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  OnSelectionEvent(WidgetSelectionEvent* aEvent,
+                   ScrollFlags aScrollFlags = ScrollFlags::None);
 
  protected:
   RefPtr<dom::Document> mDocument;
